source/lib/storage.js

```diff
@@ -74,3 +75,23 @@ export function setFavoriteBuildings(buildings: string[]) {
 export function getFavoriteBuildings(): Promise<Array<string>> {
 	return getItemAsArray(favoriteBuildingsKey)
 }
+
+/// MARK: Sis
+const courseDataKey = 'sis:course-data'
+export function setTermCourseData(term: number, courseData: Array<CourseType>) {
+	const key = courseDataKey + `:${term}:courses`
+	return setItem(key, courseData)
+}
+export function getTermCourseData(term: number): Promise<Array<CourseType>> {
+	const key = courseDataKey + `:${term}:courses`
+	return AsyncStorage.getItem(`aao:${key}`).then(stored => JSON.parse(stored))
+}
+const termInfoKey = courseDataKey + ':term-info'
+export function setTermInfo(termData: Array<TermType>) {
+	return setItem(termInfoKey, termData)
+}
+export function getTermInfo(): Promise<Array<TermType>> {
+	return AsyncStorage.getItem(`aao:${termInfoKey}`).then(stored =>
```

Comment:
You should be able to use getItemAsArray here

Reply:
oh yeah I meant to leave a note about this: for some reason if I use getItemsAsArray() and I clear the async storage, both functions return empty arrays...I think it has something to do with the fact that in getItemsAsArray() there is an "await" in front of the getItem() call. Any idea why this is happening?